src/test/fuzz/miniscript.cpp

```diff
@@ -809,11 +857,15 @@ NodeRef GenNode(F ConsumeNode, Type root_type, bool strict_valid = false) {
             // Fragment/children have not been decided yet. Decide them.
             auto node_info = ConsumeNode(type_needed);
             if (!node_info) return {};
+            if ((!miniscript::IsTapscript(SCRIPT_CTX) && node_info->fragment == Fragment::MULTI_A)
```

Comment:
I have removed this check and made the `miniscript_smart` target always generate a valid recipe.

The `SmartInfo` struct now holds two tables, one for each context. It's not ideal since there is a lot of overlap between the two sets of valid recipes, but i don't think it's worth optimizing the one-time initialization cost and small additional memory footprint at this stage. Let me know if you disagree.